### PR TITLE
Fix(UI) #8549: Fixed issue with URL in the description breaking due to characters limit.

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes :

I worked on :

- [x] Fixed issue where URL was breaking in the description preview due to character limit

- [x] Added localization in RichTextEditorPreviewer

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :
<p align="center">

Before : 
Screenshot 2022-11-09 at 3 00 33 PM

After:
Screenshot 2022-11-09 at 3 00 39 PM
Screenshot 2022-11-09 at 3 00 45 PM

Screenshot 2022-11-10 at 9 35 53 AM

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @open-metadata/ui

### CODE REVIEW

**Review by chirag-madlani — APPROVED** [review]
@Sachin-chaurasiya Can you review this once

File: openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.test.tsx [hunk 1]
```diff
@@ -209,7 +209,15 @@ describe('Test RichTextEditor Previewer Component', () => {
...
     const link = markdownParser.querySelector('a');
 
-    expect(link).toBeInTheDocument();
+    expect(link).toBeNull();
+
+    const readMoreButton = await findByTestId(container, 'read-more-button');
+
+    expect(readMoreButton).toBeInTheDocument();
+
+    fireEvent.click(readMoreButton);
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.test.tsx:218]: Wrap it inside the `act`.

File: openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx [hunk 2]
```diff
@@ -1027,3 +1027,17 @@ export const getTagValue = (tag: string | TagLabel): string | TagLabel => {
     };
   }
 };
+
+export const getTrimmedContent = (content: string, limit: number) => {
+  const slicedContent = content.slice(0, limit);
+
+  // Logic for eliminating any broken words at the end
+  // To avoid any URL being cut
+  const wordsInContent = slicedContent.split(' ');
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx:1036]:
```suggestion
  const words = slicedContent.split(' ');
```

File: openmetadata-ui/src/main/resources/ui/src/constants/constants.ts [hunk 3]
```diff
@@ -45,6 +45,7 @@ export const ADD_USER_CONTAINER_HEIGHT = 250;
 export const INGESTION_PROGRESS_START_VAL = 20;
 export const INGESTION_PROGRESS_END_VAL = 80;
 export const DEPLOYED_PROGRESS_VAL = 100;
+export const MAX_DESCRIPTION_PREVIEW_CHARACTERS = 350;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/constants/constants.ts:48]:
```suggestion
export const DESCRIPTION_MAX_PREVIEW_CHARACTERS = 350;
```

**Comment by Sachin-chaurasiya** [comment]
@aniketkatkar97 , Can you try out the changes with this description?

```https://www.google.com/imgres?imgurl=https%3A%2F%2Fstatic.clideo.com%2Ffiles%2Fcontent%2F506%2Ftwitter-meme-maker-1.png&imgrefurl=https%3A%2F%2Fclideo.com%2Fresources%2Fhow-to-make-twitter-meme&tbnid=WEpfgPjfWp6kiM&vet=12ahUKEwiKna7El6H7AhXXitgFHRNkDsQQMygCegUIARC7AQ..i&docid=r40dV35fPrPzNM&w=506&h=354&q=meme&ved=2ahUKEwiKna7El6H7AhXXitgFHRNkDsQQMygCegUIARC7AQ.
This dimension table contains the billing and shipping addresses of customers. You can join this table with the sales table to generate lists of the billing and shipping addresses. Customers can enter their addresses more than once, so the same address can appear in more than one row in this table. This table contains one row per customer address.
```

File: openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.test.tsx (hunk 1 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/common/rich-text-editor/RichTextEditorPreviewer.test.tsx]: Done

File: openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx (hunk 2 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx]: Done

File: openmetadata-ui/src/main/resources/ui/src/constants/constants.ts (hunk 3 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/constants/constants.ts]: Done

**Comment by Sachin-chaurasiya** [comment]
> > @aniketkatkar97 , Can you try out the changes with this description?
> > https://www.google.com/imgres?imgurl=https%3A%2F%2Fstatic.clideo.com%2Ffiles%2Fcontent%2F506%2Ftwitter-meme-maker-1.png&imgrefurl=https%3A%2F%2Fclideo.com%2Fresources%2Fhow-to-make-twitter-meme&tbnid=WEpfgPjfWp6kiM&vet=12ahUKEwiKna7El6H7AhXXitgFHRNkDsQQMygCegUIARC7AQ..i&docid=r40dV35fPrPzNM&w=506&h=354&q=meme&ved=2ahUKEwiKna7El6H7AhXXitgFHRNkDsQQMygCegUIARC7AQ. This dimension table contains the billing and shipping addresses of customers. You can join this table with the sales table to generate lists of the billing and shipping addresses. Customers can enter their addresses more than once, so the same address can appear in more than one row in this table. This table contains one row per customer address.
> 
> @Sachin-chaurasiya Tried with the above description, the URL got trimmed still the link worked fine. Nevertheless, I got the scenario. I have updated the function to trim the content. Can you please check that?

@aniketkatkar97, Can you share the screenshot with the above description?

**Comment by aniketkatkar97** [comment]
@Sachin-chaurasiya added the screenshot for the long URL example

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by chirag-madlani** [comment]
@aniketkatkar97 Rebase it with latest main to run cypress

File: openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx [hunk 4]
```diff
@@ -1027,3 +1027,27 @@ export const getTagValue = (tag: string | TagLabel): string | TagLabel => {
     };
   }
 };
+
+export const getTrimmedContent = (content: string, limit: number) => {
+  const lines = content.split('\n');
+  // Selecting the content in three lines
+  const contentInThreeLines = lines.slice(0, 3).join('\n');
```
[the remark this answers is not in the document]
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx:1034]: Just limiting the lines to 3 to show the read more button. As more lines can increase the height of the container. So we are checking 2 criteria, number of lines and number of characters

**Review by chirag-madlani — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**